runtime/interface.go

```diff
@@ -41,93 +43,124 @@ type Block struct {
-	// UnsafeRandom returns a random uint64, where the process of random number derivation is not cryptographically
-	// secure.
-	UnsafeRandom() (uint64, error)
-	// VerifySignature returns true if the given signature was produced by signing the given tag + data
-	// using the given public key, signature algorithm, and hash algorithm.
-	VerifySignature(
-		signature []byte,
-		tag string,
-		signedData []byte,
-		publicKey []byte,
-		signatureAlgorithm string,
-		hashAlgorithm string,
-	) (bool, error)
-	// Hash returns the digest of hashing the given data with using the given hash algorithm
-	Hash(data []byte, hashAlgorithm string) ([]byte, error)
-	// GetStorageUsed gets storage used in bytes by the address at the moment of the function call.
-	GetStorageUsed(address Address) (value uint64, err error)
-	// GetStorageCapacity gets storage capacity in bytes on the address.
-	GetStorageCapacity(address Address) (value uint64, err error)
-}
-
-type HighLevelStorage interface {
-	Interface
+	ValueExists(address Address, key []byte, caller Location) (exists bool, err error)
 
-	// HighLevelStorageEnabled should return true
-	// if the functions of HighLevelStorage should be called,
-	// e.g. SetCadenceValue
-	HighLevelStorageEnabled() bool
+	// StoredKeys returns list of keys and their sizes owned by this account
+	StoredKeys(address Address, caller Location) (keys [][]byte, sizes []uint64, err error)
+	// StorageUsed gets storage used in bytes by the address at the moment of the function call.
+	StorageUsed(address Address, caller Location) (value uint64, err error)
+	// Note: StorageCapacity has been moved to injected methods (similar to get balance)
 
-	// SetCadenceValue sets a value for the given key in the storage, owned by the given account.
-	SetCadenceValue(owner Address, key string, value cadence.Value) (err error)
+	// AddAccountKey appends a key to an account.
+	AddAccountKey(address Address, publicKey []byte, caller Location) error
+	// RemoveAccountKey removes a key from an account by index.
+	RevokeAccountKey(address Address, index uint, caller Location) error
+	// AccountPublicKey returns the account key for the given index
+	AccountPublicKey(address Address, index uint, caller Location) (publicKey []byte, err error)
+	// VerifyAccountSignature verifies a signature for the given address and index
+	// TODO RAMTIN do I need the tag here?
+	VerifyAccountSignature(address Address, index uint, signature []byte, tag string, signedData []byte, caller Location) (isValid bool, err error)
+}
+
+// Results are responsible to capture artifacts generated
+// when running the runnable
+//
+// Results won't be directly callable by users
+// and will be used by the runtime to capture outputs
+type Results interface {
+	// AppendLog appends a log to the log collection
+	AppendLog(string) error
+	// Logs returns all the logs
+	Logs() ([]string, error)
+	// returns log i of the log collection
+	Log(i uint) (string, error)
```

Comment:
This probably only makes sense if you have a function that returns the number of logs. Once that exists, `Logs()` would just be a convenience and could be removed.

Reply:
applied